tikv/server.go

```diff
@@ -562,6 +565,50 @@ func (svr *Server) UnsafeDestroyRange(context.Context, *kvrpcpb.UnsafeDestroyRan
 	return &kvrpcpb.UnsafeDestroyRangeResponse{}, nil
 }
 
+// deadlock detection related services
+// GetWaitForEntries tries to get the waitFor entries
+func (svr *Server) GetWaitForEntries(ctx context.Context,
+	req *deadlockPb.WaitForEntriesRequest) (*deadlockPb.WaitForEntriesResponse, error) {
+	// TODO
+	return &deadlockPb.WaitForEntriesResponse{}, nil
+}
+
+// Detect will handle detection rpc from other nodes
+func (svr *Server) Detect(stream deadlockPb.Deadlock_DetectServer) error {
+	for {
+		req, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		resp := &deadlockPb.DeadlockResponse{}
+		switch req.Tp {
+		case deadlockPb.DeadlockRequestType_Detect:
+			detectErr := svr.mvccStore.deadlockDetector.Detect(req.Entry.Txn, req.Entry.WaitForTxn,
+				req.Entry.KeyHash, false, nil)
+			if detectErr != nil {
+				resp.Entry = req.Entry
+				resp.DeadlockKeyHash = detectErr.KeyHash
+			} else {
+				// TODO should we just not response?
+				resp.Entry = req.Entry
```

Comment:
@coocood should we respond to caller if there is no deadlock or just let the caller waiting until timeout(3s)?

Reply:
No response if no deadlock